packages/frontend/src/redux/slices/tokenFiatValues/index.js

```diff
@@ -50,10 +38,22 @@ const tokenFiatValuesSlice = createSlice({
             builder.addCase(fetchTokenFiatValues.fulfilled, (state, action) => {
                 // Payload of .fulfilled is in the same shape as the store; just merge it!
                 // { near: { usd: x, lastUpdatedTimestamp: 1212312321, ... }
-
+                const beenUpdatedTokens = {};
+                const tokens = cloneDeep(state).tokens;
+                Object.keys(action.payload).map((token) => {
+                    const previousLastUpdatedAt = tokens[token] ? tokens[token].last_updated_at : 0;
+                    const previousPrice = tokens[token] ? tokens[token].usd : 0;
+                    const fetchedLastUpdatedAt = action.payload[token].last_updated_at;
+                    const fetchedPrice = action.payload[token].usd;
+                    if (fetchedLastUpdatedAt > previousLastUpdatedAt && fetchedPrice !== previousPrice) {
+                        beenUpdatedTokens[token] = action.payload[token];
+                    }
+                });
                 // Using merge instead of `assign()` so in the future we don't blow away previously loaded token
                 // prices when we load new ones with different token names
-                merge(state.tokens, action.payload);
+                if (Object.keys(beenUpdatedTokens).length) {
+                    merge(state.tokens, beenUpdatedTokens);
+                }
```

Comment:
I think it would be cool and cleaner to import [lodash `mergeWith`](https://lodash.com/docs/4.17.15#mergeWith) here. Something like:

```suggestion
mergeWith(state.tokens, action.payload, (previous, fetched) =>
    fetched?.last_updated_at > previous?.last_updated_at &&
    fetched?.usd != previous?.usd
        ? fetched
        : previous
);
```

I'm also somewhat concerned about things aside from `usd` changing. Maybe we can address that by importing [lodash omit](https://lodash.com/docs/4.17.15#omit) and [isEqual](https://lodash.com/docs/4.17.15#isEqual):

```suggestion
mergeWith(state.tokens, action.payload, (previous, fetched) =>
    fetched?.last_updated_at > previous?.last_updated_at &&
    !isEqual(omit(fetched, 'last_updated_at'), omit(previous, 'last_updated_at'))
        ? fetched
        : previous
);
```